src/algebraic_geometry/EllipticCurve.lean

```diff
@@ -0,0 +1,83 @@
+/-
+Copyright (c) 2021 Kevin Buzzard. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard
+-/
+
+import algebra.group_power
+import data.rat.basic
+import tactic.norm_num
+
+/-!
+# The category of elliptic curves (over a field or a PID)
+
+We give a working definition of elliptic curves which is mathematically accurate
+in many cases, and also good for computation.
+
+## Mathematical background
+
+Let `S` be a scheme. The actual category of elliptic curves over `S` is a large category,
+whose objects are schemes `E` equipped with a map `E → S`, a section `S → E`, and some
+axioms (the map is smooth and proper and the fibres are geometrically connected group varieties
+of dimension 1). In the special case where `S` is `Spec R` for some commutative ring `R`
+whose Picard group is trivial (this includes all fields, all principal ideal domains, and many
+other commutative rings) then it can be shown (using rather a lot of algebro-geometric machinery)
+that every elliptic curve is, up to isomorphism, a projective plane cubic defined by
+the equation `y^2+a₁xy+a₃y=x^3+a₂x^2+a₄x+a₆`, with `aᵢ : R`, and such that the discriminant
+of the aᵢ is a unit in `R`.
+
+More details of the construction can be found on pages 66-69 of Katz-Mazur.
```

Comment:
Do you mind adding Katz-Mazur to `docs/references.bib`?

Reply:
Done! I also added an earlier reference of Deligne.